code/modules/vore/persist/persist_vr.dm

```diff
@@ -0,0 +1,162 @@
+	persist_interround_data(occupant, pod.spawnpoint_type)
+	return 1
+
+/proc/persist_interround_data(var/mob/occupant, var/datum/spawnpoint/new_spawn_point_type)
+	ASSERT(istype(occupant))
+
+	// Find out of this mob is a proper mob!
+	if (occupant.mind && occupant.mind.loaded_from_ckey)
+		// Okay this mob has a real loaded-from-savefile mind in it!
+		var/datum/preferences/prefs = preferences_datums[occupant.mind.loaded_from_ckey]
+		if(!prefs)
+			WARNING("mind [occupant.mind] was loaded from ckey [occupant.mind.loaded_from_ckey] but no prefs datum found")
+			return
+
+		// Okay, lets do a few checks to see if we should really save tho!
+		if(!prefs.load_character(occupant.mind.loaded_from_slot))
+			WARNING("mind [occupant.mind] was loaded from slot [occupant.mind.loaded_from_slot] but loading prefs failed.")
+			return // Failed to load character
+
+		// For now as a safety measure we will only save if the name matches.
+		if(prefs.real_name != occupant.real_name)
+			log_debug("Skipping persist for [occupant] becuase [occupant.real_name] != [prefs.real_name]")
+			return
+
+		if(!prefs.persistence_settings)
+			return // Persistence disabled by preference settings
+
+		// Okay we can start saving the data
+		if(new_spawn_point_type && prefs.persistence_settings & PERSIST_SPAWN)
+			prefs.spawnpoint = initial(new_spawn_point_type.display_name)
+		if(ishuman(occupant) && occupant.stat != DEAD)
+			var/mob/living/carbon/human/H = occupant
+			testing("About to try saving stuff from [H] to [prefs] (\ref[prefs])")
+			if(prefs.persistence_settings & PERSIST_ORGANS)
+				apply_organs_to_prefs(H, prefs)
+			if(prefs.persistence_settings & PERSIST_MARKINGS)
+				apply_markings_to_prefs(H, prefs)
+			if(prefs.persistence_settings & PERSIST_WEIGHT)
+				resolve_excess_nutrition(H)
+				prefs.weight_vr = H.weight
+		prefs.save_character()
+	return
+
+// Saves mob's current organ state to prefs.
+// This basically needs to be the reverse of /datum/category_item/player_setup_item/general/body/copy_to_mob() ~Leshana
+/proc/apply_organs_to_prefs(var/mob/living/carbon/human/character, var/datum/preferences/prefs)
+	// Checkify the limbs!
+	testing("in apply_organs_to_prefs([character], \ref[prefs])")
+	for(var/name in list(BP_HEAD, BP_L_HAND, BP_R_HAND, BP_L_ARM, BP_R_ARM, BP_L_FOOT, BP_R_FOOT, BP_L_LEG, BP_R_LEG, BP_GROIN, BP_TORSO))
+		var/obj/item/organ/external/O = character.organs_by_name[name]
+		if(!O)
+			prefs.organ_data[name] = "amputated"
+		else if(O.robotic >= ORGAN_ROBOT)
+			prefs.organ_data[name] = "cyborg"
+			if(O.model)
+				prefs.rlimb_data[name] = O.model
+			else
+				prefs.rlimb_data.Remove(name) // Missing rlimb_data entry means default model
+		else
+			prefs.organ_data.Remove(name) // Misisng organ_data entry means normal
+
+	// Internal organs also
+	for(var/name in list(O_HEART,O_EYES,O_LUNGS,O_BRAIN))
+		var/obj/item/organ/I = character.internal_organs_by_name[name]
+		if(I)
```

Comment:
My only comment about this section is that it should probably use species.has_organ for internal, and species.has_limbs for body parts, because some species have weird limb setups. ALSO if you include appendix that would keep people from getting appendicitis twice due to persistence!

Keep in mind that might not be exactly perfect due to how synths work with robotizing things. Does O_HEART refer to the cell, etc?